scripts/validate_docstrings.py

```diff
@@ -331,6 +336,41 @@ def parameter_mismatches(self):
 
         return errs
 
+    @property
+    def pep8_violations(self):
+        with self._file_representation() as file:
+            application = Flake8()
+            application.initialize(["--doctests", "--quiet"])
+            application.run_checks([file.name])
+            application.report()
+            stats = application.guide.stats
+            return [
+                "{} {} {}".format(s.count, s.error_code, s.message)
+                for s in stats.statistics_for('')
+            ]
```

Comment:
I'd personally prefer to `yield from application.guide.stats.statistics_for('')` instead of returning the string. In case the information is used in more than one place, having the components can be useful.

Also, it's an opinion, but I wouldn't use a property here, as this feels more like an action than an attribute. A method `validate_pep8` seems easier to understand.